lib/src/arithmetic/values/floating_point_values/floating_point_value.dart

```diff
@@ -310,19 +338,101 @@ class FloatingPointValue implements Comparable<FloatingPointValue> {
         doubleVal = (sign.toBool() ? -1.0 : 1.0) *
             pow(2.0, minExponent) *
             mantissa.toBigInt().toDouble() /
-            pow(2.0, mantissa.width);
+            pow(2.0, mantissa.width - (explicitJBit ? 1 : 0));
       } else if (!isNaN) {
         doubleVal = (sign.toBool() ? -1.0 : 1.0) *
-            (1.0 + mantissa.toBigInt().toDouble() / pow(2.0, mantissa.width)) *
+            ((explicitJBit ? 0.0 : 1.0) +
+                mantissa.toBigInt().toDouble() /
+                    pow(2.0, mantissa.width - (explicitJBit ? 1 : 0))) *
             pow(2.0, exponent.toInt() - bias);
       }
     }
     return doubleVal;
   }
 
+  /// Convert to a [FloatingPointValue] with a mantissa that is one smaller
+  /// due to the implicit J-bit.
+  FloatingPointValue toFloatingPointValue() {
```

Comment:
the name and doc of this is confusing now -- is this actually `toImplicitJBitFloatingPointValue` or something?  And only reduces mantissa width if it was previously explicit?  also, this is not using the populator for potential subclasses.

should this really just be deleted and just leave `canonicalize`?

Reply:
So the potentially cleaner way to do this is to:
.  make an ofFloatingPointValue which does the correct thing based on the input type and the populator type.
.  if they are both the same type, just clone
. Have a flag that says alwaysCanonicalize if they are the same explicit type.

Reply:
Actually:  a new proposal:
```dart
({s,e,m}) _components(fpv, {canonicalize=true}) {}
ofFloatingPointValue(fpv, {alwaysCanonicalize=false})  --- calls _components(fpv, canonicalize: alwaysCanonicalize);
```
This way, we can convert from I to E or E to I or E to E with/without canonicalization.

Reply:
So I resolved to use
```dart
FloatingPointValuePopulator.ofFloatingPointValue(FloatingPointValue fpv, {bool canonicalizeExplicit=false});
```
since it is one routine to do all conversions.  It calls _components, which also has a canonicalize flag.